From 6f65ec262f7795d1b72325bbba84511c3b525b9f Mon Sep 17 00:00:00 2001
From: Badhri Jagan Sridharan <Badhri@google.com>
Date: Mon, 15 Dec 2014 16:42:27 -0800
Subject: [PATCH 526/634] usb: gadget: configfs: Add usb_function ptr to fi
 struct

Add a pointer to the usb_function inside the
usb_function_instance structure to service
functions specific setup requests even before
the function gets added to the usb_gadget

Signed-off-by: Badhri Jagan Sridharan <Badhri@google.com>
Change-Id: I6f457006f6c5516cc6986ec2acdf5b1ecf259d0c
---
 include/linux/usb/composite.h | 1 +
 1 file changed, 1 insertion(+)

diff --git a/include/linux/usb/composite.h b/include/linux/usb/composite.h
index dba63f5..dcae90e 100644
--- a/include/linux/usb/composite.h
+++ b/include/linux/usb/composite.h
@@ -468,6 +468,7 @@ struct usb_function_instance {
 	struct config_group group;
 	struct list_head cfs_list;
 	struct usb_function_driver *fd;
+	struct usb_function *f;
 	int (*set_inst_name)(struct usb_function_instance *inst,
 			      const char *name);
 	void (*free_func_inst)(struct usb_function_instance *inst);
-- 
1.9.1

